src/pikachu/sm/pikachuApi.ts

```diff
@@ -0,0 +1,26 @@
+import pokemonApi from 'pokemon/sm/pokemonApi';
+import createRtkQueryFetch from 'reduxToolkit/util/createRtkQueryFetch';
+
+const pikachuApi = pokemonApi.injectEndpoints({
+  endpoints: (build) => ({
+    getPikachu: build.query<pokemon.Pokemon, 'reduxSaga' | 'rtkQueryHook'>({
+      query: (method) => {
+        return {
+          url: `pokemon/pikachu?method=${method}`,
+          method: 'get',
+        };
+      },
+    }),
+  }),
+});
+
+export const {
+  useGetPikachuQuery,
+} = pikachuApi;
+
+/**
+ * `async...await` conversion.
+ */
+export const getPikachu = createRtkQueryFetch.query(pikachuApi.endpoints.getPikachu.initiate);
```

Comment:
One issue discovered about using RTK Query in a typical Redux Saga HTTP request flow is that the API endpoint methods generated through the RTK Query builder are not immediately usable.

To put it in perspective, this is what I'm used to seeing:

```ts
function* getPikachu() {
  return yield call(fetchWithCustomLogic, '/pikachu');
}

function* getPikachuTask() {
  try {
    const data = yield call(getPikachu);
    yield put(pikachuActions.getPikachuDone(data));
  } catch (error) {
    yield put(pikachuActions.getPikachuError(error));
  }
}
```

Swapping `getPikachu()` with `pikachuApi.endpoints.getPikachu.initiate()` without any wrappings is not viable!

Assuming `pikachuApi.endpoints.getPikachu.initiate()` accepts zero parameters...

```ts
import store from 'root/sm/store';

function* getPikachuTask() {
  try {
    // This doesn't work.
    // The .getPikachu.initiate() method is a Redux Thunk action creator. You need to run store.dispatch() to invoke the request.
    const data = yield call(pikachuApi.endpoints.getPikachu.initiate);

    // This doesn't work, either.
    // The .dispatch() method does return a Promise which in theory, can be 
    const data = yield call(dispatch(pikachuApi.endpoints.getPikachu.initiate()));

    // This works but with some problems.
    // Using this approach allows the action to be dispatched and the endpoint to be called under the hood.
    // However, this creates a non-blocking call, which in the case of this flow, is incorrect behavior, because it requires a blocking call instead.
    yield put(pikachuApi.endpoints.getPikachu.initiate());

    // This works.
    // However, it has to be rewrapped into a function to return a Promise. Further implementation is required to extract data.
    yield call(() => dispatch(pikachuApi.endpoints.getPikachu.initiate()));
    // If extracting the .data portion from the Promise object returned.
    const data = yield call(async () => (await dispatch(pikachuApi.endpoints.getPikachu.initiate())).data);

    yield put(pikachuActions.getPikachuDone(data));
  } catch (error: any) {
    yield put(pikachuActions.getPikachuError(error));
  }
}
```